### PR TITLE
Use logger formatter instead of f-strings in log statements

### PR DESCRIPTION
**Pull request by mrkickling**
This is the recommended way to do logging by my linter (I believe for optimization reasons), and it looks nice in my opinion.

### CODE REVIEW

File: maltoolbox/language/languagegraph.py [hunk 1]
```diff
@@ -470,8 +474,10 @@ def process_step_expression(self,
 
         if logger.isEnabledFor(logging.DEBUG):
             # Avoid running json.dumps when not in debug
-            logger.debug('Processing Step Expression:\n' \
-                + json.dumps(step_expression, indent = 2))
+            logger.debug(
+                'Processing Step Expression:\n %s',
```
Comment (nkakouros) [inline maltoolbox/language/languagegraph.py:478]:
```suggestion
                'Processing Step Expression:\n%s',
```

File: maltoolbox/language/languagegraph.py [hunk 2]
```diff
@@ -930,20 +963,26 @@ def _get_associations_for_asset_type(self, asset_type: str) -> dict:
...
+        )
         associations = []
 
         asset = next((asset for asset in self._lang_spec['assets'] if asset['name'] == \
             asset_type), None)
         if not asset:
-            logger.error(f'Failed to find asset type {asset_type} when '\
-                'looking for associations.')
+            logger.error(
+                'Failed to find asset type %s when '\
```
Comment (nkakouros) [inline maltoolbox/language/languagegraph.py:976]:
```suggestion
                'Failed to find asset type %s when '
```

File: maltoolbox/model.py [hunk 3]
```diff
@@ -85,15 +85,17 @@ def add_asset(
                 if allow_duplicate_names:
                     asset.name = asset.name + ':' + str(asset.id)
                 else:
-                    raise ValueError(f'Asset name {asset.name} is a '
-                    'duplicate and we do not allow duplicates.')
+                    raise ValueError(
+                        f'Asset name {asset.name} is a duplicate'
```
Comment (nkakouros) [inline maltoolbox/model.py:89]: Still f-string.

**Review by nkakouros — COMMENTED** [review]
I think this is a great opportunity to setup some CI checks. For instance, ruff will [pick up](https://docs.astral.sh/ruff/rules/logging-f-string/) f-strings in logging statements and can help check if you missed some.

File: maltoolbox/model.py (hunk 3 above)
Comment (mrkickling) [reply maltoolbox/model.py:89]: I focused on log-statements in this PR, not exceptions. I didn't get any warnings from my linter for f-strings in exceptions, so I let them be.

Comment (nkakouros) [reply maltoolbox/model.py:89]: oh right, false alarm.

File: maltoolbox/language/languagegraph.py (hunk 1 above)
Comment (mrkickling) [reply maltoolbox/language/languagegraph.py]: Fixed

File: maltoolbox/language/languagegraph.py (hunk 2 above)
Comment (mrkickling) [reply maltoolbox/language/languagegraph.py]: Fixed

**Review by nkakouros — APPROVED** [review]

**Pull request merged**